src/types.rs

```diff
@@ -261,3 +262,59 @@ impl BufRingEntry {
         &(*ring_base).0.resv
     }
 }
+
+/// A destination slot for sending fixed resources
+/// (e.g. [`opcode::MsgRingSendFd`](crate::opcode::MsgRingSendFd)).
+#[derive(Debug, Clone, Copy)]
+pub struct DestinationSlot {
+    dest: KernelIndexDest,
+}
+
+#[derive(Debug, Clone, Copy)]
+enum KernelIndexDest {
```

Comment:
This enum seems unnecessary, keeping u32 is fine.

Reply:
I would have suggested SlotSelection at this point.

But you need @quininer's vote so you can ignore me and use u32. I will stop giving opinions here.

Reply:
I'm appreciate the reviews from both of you so far :smile: 

I can drop the enum and directly keep the numerical representation. Would you maybe want a `NonZeroU32` here? (the `0` is not possible due to +1 shift enforced by the constructor)

Reply:
`NonZeroU32` sounds good.